src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -88,5 +89,77 @@ IEnumerator<int> IEnumerable<int>.GetEnumerator()
 
         IEnumerator IEnumerable.GetEnumerator()
             => GetEnumerator();
+
+        public bool Contains(int value)
+        {
+            if (!IsBound) throw new InvalidOperationException("Unbound ranges cannot contain");
```

Comment:
If it's unbound, then it contains any value. If it has unbound start, check the end only. Same for ranges with unbound start.

```csharp
if (IsBound) return Start <= value && value < End;
if (Start == UnboundStart) return value < End;
if (End == UnboundEnd) return Start <= value;
return false;
```

Reply:
Do not forget to check direction.

Reply:
Theoretically yes, but what are the scenarios for calling Contains on unbound ranges?

Reply:
Exceptions are okay for `GetEnumerator` since we can't iterate over infinity sequences, but `Contains` can work and therefore should. Otherwise, it wouldn't be useful, and developers would be managed to write they own implementation. For example:

```csharp
class Example
{
    private Range _range;

    public Example(Range range) => _range = range;

    public void DoSomething(int value)
    {
        if (_range.Contains(value))
        {
        }
        else
        {
        }
    }
}
```

What if `range` provided to the constructor is unbound? Should I use `(_range.Start == Range.UnboundStart || _range.Start <= value) && (_range.End == Range.UnboundEnd || value < _range.End)` to know if the `value` is in `_range` to be sure that the code will not throw?

Reply:
Also the `in` pattern matching will use `Contains`.

```csharp
switch (variable)
{
    case in ..0: // should not throw
        break;
    // ...
    case in 5..: // should not throw
        break;
}
```